goci-interfaces/goci-ui/src/main/java/uk/ac/ebi/spot/goci/ui/controller/GeneImpcLinkController.java

```diff
@@ -0,0 +1,55 @@
+package uk.ac.ebi.spot.goci.ui.controller;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import uk.ac.ebi.spot.goci.ui.component.ImpcIdentifierMap;
+import uk.ac.ebi.spot.goci.ui.constants.SearchUIConstants;
+import uk.ac.ebi.spot.goci.ui.service.GeneImpcLinkService;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping(value = SearchUIConstants.API_V1 + SearchUIConstants.GENE_URI)
+public class GeneImpcLinkController {
+    private static final Logger log = LoggerFactory.getLogger(GeneImpcLinkController.class);
+
+    @Autowired
+    ImpcIdentifierMap impcIdentifierMap;
+
+    @Autowired
+    GeneImpcLinkService geneImpcLinkService;
+
+    /** API called UI to retreive Mouse Ortholog mapping for Gene
+     *
+     * @param geneId
+     * @return
+     */
+    @ResponseStatus(HttpStatus.OK)
+    @GetMapping(value = "{geneId}/impclink")
```

Comment:
This endpoint resolves to /v1/genes/{geneId}/impclink 

We could rename it to: /v1/impc-mouse-orthologs/{geneId}

We have resources named gene in the catalog at the moment accessible at "https://www.ebi.ac.uk/gwas/genes" 

First the /gene endpoint will conflict with the existing gene resource especially for any future maintainer of the project 

Secondly, We should name the endpoint using the plural noun of the actual resource been retrieved which is "IMPC Mouse Orthologs"

Hence the justification for the endpoint: /v1/impc-mouse-orthologs/{geneId}